src/setup_node_env/node_version_validator.test.js

```diff
@@ -51,31 +51,33 @@ describe('NodeVersionValidator', function () {
     );
   });
 
-  it('should run the script WITH error if the major version is higher', function (done) {
-    testValidateNodeVersion(done, requiredNodeVersionWithDiff(+1, 0, 0), true);
-  });
-
-  it('should run the script WITH error if the major version is lower', function (done) {
-    var lowerMajorVersion = requiredNodeVersionWithDiff(-1, 0, 0);
-    testValidateNodeVersion(
-      done,
-      lowerMajorVersion,
-      REQUIRED_NODE_JS_VERSION !== lowerMajorVersion
-    );
-  });
-
-  it('should run the script WITH error if the minor version is higher', function (done) {
-    testValidateNodeVersion(done, requiredNodeVersionWithDiff(0, +1, 0), true);
-  });
-
-  it('should run the script WITH error if the minor version is lower', function (done) {
-    var lowerMinorVersion = requiredNodeVersionWithDiff(0, -1, 0);
-    testValidateNodeVersion(
-      done,
-      lowerMinorVersion,
-      REQUIRED_NODE_JS_VERSION !== lowerMinorVersion
-    );
-  });
+  if (!process.env.OSD_NODE_HOME) {
```

Comment:
are you actually able to do something like this: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/cli_plugin/install/settings.test.js#L78

which is saving the stored env variable, then modify the environment variable and right specific tests related to if OSD_NODE_HOME is set?

Reply:
About the tests for OSD_NODE_HOME issue - in the link that you've attached with the comment, the way environment variables are set is valid only for the scope of the function in which they're set. In case of tests for OSD_NODE_HOME, the scope is changing hence that method of setting or unsetting the environment variables is not working. That was the reason why I didn't add tests. Please let me know if there is a way to directly set and unset environment variables through node in the system and not just for the scope of the function. Or if there is a way to add these test cases without doing that.

Reply:
These unit tests are just for the internal logic of `node_version_validator.js`, so it's fine to essentially mock the return value of the env variable. I think it should be possible to just mock `process.env` if need be.

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/75b3ca8271556f9b40950ac53cd58f4e4783e733/src/setup_node_env/node_version_validator.test.js#L94

Because of this line, I think mocking` process.env` will not work. The file node version validator is running in a different process from test, so `process.env` for the test is not applicable for node version validator in this case.

Reply:
Is this thread outdated? Seems like there are test cases now.

Reply:
Yes the test is updated and we are using `Object.defineProperty` to modify the `process` env variable.